### DISCUSSION TITLE
Is KNative + Istio optional?

### DISCUSSION THREAD

**Discussion by iciclespider** [kubeflow/kfserving#3494] [Q&A] [answered]
This image https://kserve.github.io/website/0.11/images/kserve_layer.png says: KNative + Istio (Serverless Layer- Optional)

Is this in fact the case, and if so are the deployment examples?

**Comment by israel-hdez** [comment]
The Kubernetes Deployment (aka RawDeployment) is the installation that won't require KNative nor Istio: https://kserve.github.io/website/master/admin/kubernetes_deployment/

**Reply by iciclespider** [reply]
That is my confusion. According to the link you provided, it says to install Istio.

**Reply by israel-hdez** [reply, marked as answer]
There is a note mentioning that Istio is being used just as an _ingress controller_. If you have some other ingress controller, you can skip Istio installation and set the `ingressClassName` accordingly in the `inferenceservice-config` ConfigMap (see step 3.ii of the link I provided).

AFAIK, in RawDeployment mode, KServe will create an [standard Ingress resource](https://kubernetes.io/docs/concepts/services-networking/ingress/#the-ingress-resource) using the configured ingressClassName. It shouldn't create any Istio-specific resource. I guess Istio is mentioned only because it is the controller used for testing. But as long as you have something that reconciles standard Kubernetes Ingress resources, you should be fine.

**Reply by iciclespider** [reply]
Excellent! Thanks for the clarification.
